### PR TITLE
Module 8: UI Task

### PR DESCRIPTION
**Pull request by Juandoc17**
UI task implemented

### CODE REVIEW

**Comment by KorolIvan** [comment]
I guess this task wasn't clear for you. Looking on your PR I understood that you try to use absolutely different framework that is has nothing to java and related to native/pure JS.
If you will take a look on the file (https://epam-my.sharepoint.com/:w:/p/yury_karpinski/EZ2CoYAxM7dNmZ99phKucwQBbjKWWeADt8VMFTjQZMPh0w?e=kNAQkE) more closely you will find that there is some tips already added to the descriptions:
task #1: Alternative tool -> you using pure selenium, try to add selenide (done particular)
task #2: ACTIONS. JS EXECUTOR -> you did add the native JS testing framework (cypress) that is a bit wrong in my opinion as you start build the TAF on the Java.In the task asking you to use Actions (https://www.selenium.dev/selenium/docs/api/java/org/openqa/selenium/interactions/Actions.html , https://www.selenium.dev/documentation/webdriver/actions_api/) and JS Executor (https://www.guru99.com/execute-javascript-selenium-webdriver.html , https://www.selenium.dev/selenium/docs/api/java/org/openqa/selenium/JavascriptExecutor.html). Please make the changes with required approaches.

P.S. You can find all information into the provided links, all you need it just make the realization. ))
Have fun.))

File: src/test/java/org/example/HomePageTests.java [hunk 1]
```diff
@@ -68,5 +68,37 @@ public void validateTagTypesOfCriticalElements() {
 		assertEquals(homePage.getCheckBoxesElementTag(), "a");
 		logger.info("All Elements with correct Tag Types");
 	}
+	@Test
+    @DisplayName("Validation of Drag and Drop Functionality")
+    public void validateDragAndDropFunctionality() {
```
Comment (KorolIvan) [inline src/test/java/org/example/HomePageTests.java]: Please add assertion that the element/whatever appears after the action completed.

File: src/test/java/org/example/HomePageTests.java [hunk 2]
```diff
@@ -68,5 +68,37 @@ public void validateTagTypesOfCriticalElements() {
...
+        logger.info("Assert Drag and Drop Functionality");
+        homePage.dragAndDrop(homePage.getDashboardHeader(), homePage.getLoginLink());
+        logger.info("Drag and Drop functionality validated");
+    }
+
+    @Test
+    @DisplayName("Validation of Resize Element Functionality")
+    public void validateResizeElementFunctionality() {
```
Comment (KorolIvan) [inline src/test/java/org/example/HomePageTests.java]: Ditto.

File: src/test/java/org/example/HomePageTests.java [hunk 3]
```diff
@@ -68,5 +68,37 @@ public void validateTagTypesOfCriticalElements() {
...
+    public void validateScrollToElementFunctionality() {
+        logger.info("Assert Scroll To Element Functionality");
+        homePage.scrollToElement(homePage.getLoginLink());
+        assertTrue(homePage.isElementInView(homePage.getLoginLink()), "Element is not in view.");
+        logger.info("Scroll To Element functionality validated");
+    }
+
+    @Test
+    @DisplayName("Validation of Click Using JS Functionality")
+    public void validateClickUsingJSFunctionality() {
```
Comment (KorolIvan) [inline src/test/java/org/example/HomePageTests.java]: Ditto.

File: src/test/java/org/example/selenide/HomePageTest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        Configuration.reportsFolder = "test-result/reports";
+    }
+
+    @Test
+    public void testHomePage() {
+        open("https://the-internet.herokuapp.com/");
+
+        try {
+            //homePage.dashboardHeader.shouldBe(visible);
+            //assertTrue(homePage.isDashboardDisplayed());
```
Comment (KorolIvan) [inline src/test/java/org/example/selenide/HomePageTest.java]: Please, provide some test realization that is throwing the exception/error to check if the screenshot will be saved. If I right understood you have it as a plan, am I right?

File: package-lock.json
Comment (KorolIvan) [inline package-lock.json:1]: I believe that this file can be removed as redundant.

File: package.json
Comment (KorolIvan) [inline package.json:1]: I believe that this file can be removed as redundant.

**Review by KorolIvan — COMMENTED** [review]
Please take a look on comments and provide some solution. Also make code formatting through the project and align the code (Ctrl + Alt + L -> Idea shortcut).

File: package.json
Comment (Juandoc17) [reply package.json:1]: File removed

File: package-lock.json
Comment (Juandoc17) [reply package-lock.json:1]: File removed

File: src/test/java/org/example/HomePageTests.java (hunk 1 above)
Comment (Juandoc17) [reply src/test/java/org/example/HomePageTests.java]: Assertion implemented!

File: src/test/java/org/example/HomePageTests.java (hunk 2 above)
Comment (Juandoc17) [reply src/test/java/org/example/HomePageTests.java]: Assertion implemented!

File: src/test/java/org/example/HomePageTests.java (hunk 3 above)
Comment (Juandoc17) [reply src/test/java/org/example/HomePageTests.java]: Assertion implemented!

File: src/test/java/org/example/selenide/HomePageTest.java (hunk 4 above)
Comment (Juandoc17) [reply src/test/java/org/example/selenide/HomePageTest.java]: Yes! I forgot this commented code, now is fixed!

**Review by KorolIvan — APPROVED** [review]
LGTM/Approved. Good job.

**Pull request merged**